esx_service/vmdk_ops.py

```diff
@@ -569,9 +569,26 @@ def findDeviceByPath(vmdk_path, vm):
     return None
 
 
-def busInfo(unit_number, bus_number):
+def get_controller_pci_slot(vm, pvscsi, key_offset):
+    if pvscsi[0].slotInfo:
+       return(pvscsi[0].slotInfo.pciSlotNumber)
+    else:
+       # Slot number is got from from the VM config
+       key = 'scsi{0}.pciSlotNumber'.format(pvscsi[0].key -
```

Comment:
This is what I discovered. VSphere 6.0 VMODL is different from the current
one and since we support both earlier and later VSphere releases need to
handle both cases. If the slotInfo is set then its the newer VMODL else the
old one.

On Mon, Jul 18, 2016 at 11:10 PM, Andrew J. Stone <notifications@github.com>
wrote:

> In esx_service/vmdk_ops.py
> <https://github.com/vmware/docker-volume-vsphere/pull/534#discussion_r71193155>
> :
>
> > @@ -569,9 +569,26 @@ def findDeviceByPath(vmdk_path, vm):
> >      return None
> >
> >
> > -def busInfo(unit_number, bus_number):
> > +def get_controller_pci_slot(vm, pvscsi, key_offset):
> > +    if pvscsi[0].slotInfo:
> > +       return(pvscsi[0].slotInfo.pciSlotNumber)
> > +    else:
> > +       # Slot number is got from from the VM config
> > +       key = 'scsi{0}.pciSlotNumber'.format(pvscsi[0].key -
>
> When would pvscsi[0].slotInfo not exist, but pvscsi[0].key would exist?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vmware/docker-volume-vsphere/pull/534/files/8d1b6f9ffa63e607a897da8cecde4d4b1630b83c#r71193155>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/APHseJWTOi8QLyEDPa75z9s-ok2B8Sw4ks5qW7qjgaJpZM4JLQN3>
> .
>